hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystem.java

```diff
@@ -615,10 +615,17 @@ protected void testListStatusOnLargeDirectory() throws Exception {
     ArrayList<String> actualPathList = new ArrayList<>();
     if (rootItemCount != fileStatuses.length) {
       for (int i = 0; i < fileStatuses.length; i++) {
-        actualPaths.add(fileStatuses[i].getPath().getName());
+        boolean duplicate =
+                actualPaths.add(fileStatuses[i].getPath().getName());
+        if (!duplicate) {
+          LOG.info("Duplicate path:{} in FileStatusList",
```

Comment:
Do we expect Duplicate paths in ListStatus ?

Reply:
Thanks @mukul1987 for the reviews. No, there shouldn't be any duplicated path in the list, but I suspected some duplicate elements which makes the test case failure. So, I have added logs to capture this part.

In the latest run, its again failed and from the logs I saw that 5010 has been duplicated. I am checking it further to find out the cause. Interestingly it is always passing locally and failed to re-produce it. Do you have any thoughts?
```
2021-01-18 13:46:39,800 [Listener at 127.0.0.1/41009] INFO  ozone.TestOzoneFileSystem (TestOzoneFileSystem.java:testListStatusOnLargeDirectory(621)) - Duplicate path:5110 in FileStatusList
2021-01-18 13:46:39,802 [Listener at 127.0.0.1/41009] INFO  ozone.TestOzoneFileSystem (TestOzoneFileSystem.java:testListStatusOnLargeDirectory(627)) - actualPathsSize: 5111
2021-01-18 13:46:39,802 [Listener at 127.0.0.1/41009] INFO  ozone.TestOzoneFileSystem (TestOzoneFileSystem.java:testListStatusOnLargeDirectory(628)) - actualPathListSize: 5112
2021-01-18 13:46:39,809 [Listener at 127.0.0.1/41009] INFO  ozone.TestOzoneFileSystem (TestOzoneFileSystem.java:testListStatusOnLargeDirectory(631)) - actualPaths: []
2021-01-18 13:46:39,810 [Listener at 127.0.0.1/41009] INFO  ozone.TestOzoneFileSystem (TestOzoneFileSystem.java:testListStatusOnLargeDirectory(632)) - actualPathList: []
```

Reply:
Test is creating 0 to 5110 directories and `5110` directory is the last one. 
I am suspecting that `5110` exists in table cache and its again visible while iterating over DB, do you think so?

Reply:
I have updated the patch by using two separate maps - cacheDirMap and cacheFileMap to make unique OmKeyInfos. I have got a green build now.